src/crypto-api.ts

```diff
@@ -284,6 +284,28 @@ export interface CryptoApi {
      * @returns a VerificationRequest when the request has been sent to the other party.
      */
     requestDeviceVerification(userId: string, deviceId: string): Promise<VerificationRequest>;
+
+    /**
+     * Fetches the backup private key, if cached
+     * @returns the key, if any, or null
+     */
+    getSessionBackupPrivateKey(): Promise<Uint8Array | null>;
+
+    /**
+     * Stores the session backup key to the cache
+     * @param key - the private key
+     * @returns a promise so you can catch failures
+     */
+    storeSessionBackupPrivateKey(key: ArrayLike<number>): Promise<void>;
+
+    /**
+     * sign the given object with our ed25519 key
+     *
+     * @param obj -  Object to which we will add a 'signatures' property
+     */
+    signObject<T extends ISignableObject & object>(obj: T): Promise<void>;
```

Comment:
do you think we actually need this? I note that the only place it is called (`MatrixClient.createKeyBackupVersion`) has this comment:

```
This can probably go away very soon in
favour of just signing with the cross-singing master key.
```

is "very soon" less than 3.5 years?

Reply:
mm, not sure about the comment. What we will stop doing is trusting a backup if it's signed by a device we trust. But I think signing with our own device is a way to mark as trusted.